ArcGISRuntimeSDKQt_CppSamples/Geometry/GeodesicOperations/GeodesicOperations.cpp

```diff
@@ -0,0 +1,123 @@
+
+GeodesicOperations::GeodesicOperations(QQuickItem* parent /* = nullptr */):
+  QQuickItem(parent)
+{
+}
+
+void GeodesicOperations::init()
+{
+  // Register the map view for QML
+  qmlRegisterType<MapQuickView>("Esri.Samples", 1, 0, "MapView");
+  qmlRegisterType<GeodesicOperations>("Esri.Samples", 1, 0, "GeodesicOperationsSample");
+}
+
+void GeodesicOperations::componentComplete()
+{
+  QQuickItem::componentComplete();
+
+  // find QML MapView component
+  m_mapView = findChild<MapQuickView*>("mapView");
+
+  // Create a map using the imagery basemap
+  m_map = new Map(Basemap::imagery(this), this);
+
+  // Set map to map view
+  m_mapView->setMap(m_map);
+
+  // Create a GraphicsOverlay
+  GraphicsOverlay* graphicsOverlay = new GraphicsOverlay(this);
+  m_mapView->graphicsOverlays()->append(graphicsOverlay);
+
+  // Create Graphic Symbols
+  SimpleMarkerSymbol* markerSymbol = new SimpleMarkerSymbol(SimpleMarkerSymbolStyle::Circle, QColor("blue"), 10.0f /*size*/, this);
+  SimpleLineSymbol* pathSymbol = new SimpleLineSymbol(SimpleLineSymbolStyle::Dash, QColor("blue"), 5.0f /*width*/, this);
+
+  // Create NYC graphic
+  const Point nycPoint(-73.7781, 40.6413, SpatialReference(4326));
+  m_nycGraphic = new Graphic(nycPoint, markerSymbol, this);
+  graphicsOverlay->graphics()->append(m_nycGraphic);
+
+  // Create destination graphic
+  m_destinationGraphic = new Graphic(this);
+  m_destinationGraphic->setSymbol(markerSymbol);
+  graphicsOverlay->graphics()->append(m_destinationGraphic);
+
+  // Create path graphic
+  m_pathGraphic = new Graphic(this);
+  m_pathGraphic->setSymbol(pathSymbol);
+  graphicsOverlay->graphics()->append(m_pathGraphic);
+
+  // connect to mouse clicked signal
+  connect(m_mapView, &MapQuickView::mouseClicked, this, [this, nycPoint](QMouseEvent& mouseEvent)
+  {
+    // re-project the point to match the NYC graphic
+    const Point clickedPoint = m_mapView->screenToLocation(mouseEvent.x(), mouseEvent.y());
+    const Point destination = GeometryEngine::project(clickedPoint, m_nycGraphic->geometry().spatialReference());
+
+    // update the destination graphic
+    m_destinationGraphic->setGeometry(destination);
+
+    // create line with start/end points
+    const QList<Point> points = {nycPoint, destination};
+    const Polyline polyline = pointsToPolyline(points);
+
+    // densify the path as a geodesic curve and show it with the path graphic
+    constexpr double maxSegmentLength = 1.0;
+    const LinearUnit unitOfMeasurement(LinearUnitId::Kilometers);
+    constexpr GeodeticCurveType curveType = GeodeticCurveType::Geodesic;
+    const Geometry pathGeometry = GeometryEngine::densifyGeodetic(polyline, maxSegmentLength, unitOfMeasurement, curveType);
+
+    // update the graphic
+    m_pathGraphic->setGeometry(pathGeometry);
+
+    // calculate the path's geodetic length
+    m_distanceText = QString::number((int)GeometryEngine::lengthGeodetic(pathGeometry, unitOfMeasurement, curveType));
```

Comment:
@ldanzinger Why cast to `int` here instead of just converting the double to string?